M2/Macaulay2/packages/AbstractSimplicialComplexes.m2

```diff
@@ -71,236 +76,194 @@ AbstractSimplicialComplex.GlobalAssignHook = globalAssignFunction
 
---- Constructors for AbstractSimplicialComplexs
+--- Constructors for AbstractSimplicialComplexs ---
 
 abstractSimplicialComplex = method()
 
--- the most basic constructor of a AbstractSimplicialComplex
-
--- The idea is to make a simplical complex starting from a list of faces.
+--  What follows is the most basic constructor of AbstractSimplicialComplex.
+--  The idea is to make a simplical complex starting from a list of faces.
 --  The list of faces need not be facets.
---  The constructor returns the simplicial complex (with all of its faces) that is
---  generated by this list of faces 
---  By default, it is assumed that the kfaces are all lex ordered positive integers 
+--  The constructor returns the simplicial complex (with all of its faces)
+--  that is generated by this list of faces. 
+--  By default, it is assumed that the kfaces are all lex ordered
+--  positive integers.
 
 makeKFaces := (L,k) -> (
-    toList(set(flatten(apply(#L, i -> subsets(sort L_i,k)))))
-    )
+     toList(set(flatten(apply(#L, i -> subsets(sort L_i,k))))))
 
 makeAllFaces := (L) -> (
-    numberOfFaces := #L;
---  find the highest dimensional face    
-    n := max(apply(numberOfFaces, i-> # (L_i)));
-    flatten(for k from 0 to n list {k-1 => sort makeKFaces(L,k)})
-    )
-
+     numberOfFaces := #L;
+     n := max(apply(numberOfFaces, i-> # (L_i))); --  find the highest face dimension    
+     flatten(for k from 0 to n list {k-1 => sort makeKFaces(L,k)}))
 
 abstractSimplicialComplex(List) := AbstractSimplicialComplex => L -> (
-    return new AbstractSimplicialComplex from makeAllFaces(L)
-    )
+     return new AbstractSimplicialComplex from makeAllFaces(L))
 
+--- The following method will make the (n-1)-dimensional
+--  n-simplex on [n] = {1,...,n}.
 
---- The following method will make the (n-1)-dimensional n-simplex on [n] = {1,...,n}
 abstractSimplicialComplex(ZZ) := AbstractSimplicialComplex => (n) -> (
-    L := for i from 1 to n list i;
-    return abstractSimplicialComplex({L})
-    )
+     L := for i from 1 to n list i;
+     return abstractSimplicialComplex({L}))
 
---- Make the "r-skeleton" on [n] = {1,...n}
+--- Make the "r-skeleton" on [n] = {1,...n}.
 
 abstractSimplicialComplex(ZZ,ZZ) := AbstractSimplicialComplex => (n,r) -> (
-    return abstractSimplicialComplex subsets(for i from 1 to n list i,r)
-    )
+     return abstractSimplicialComplex subsets(for i from 1 to n list i,r))
 
---
-
--- making random simplicial complexes --
+----------------------------------------
+-- Making random simplicial complexes --
+----------------------------------------
 
--- make a random subset of {1,...,n}
+-- Make a random subset of {1,...,n}.
 
 randomSubset = method()
 
 randomSubset(ZZ) := List => (n) -> (
-   setRandomSeed(currentTime());
-   k := random(1,n); 
-   sort unique (for i from 1 to k list (random(1,n)))
-    )
+     setRandomSeed(currentTime()); -- without doing this we will obtain very biased output
```

Comment:
I wouldn't call `setRandomSeed` in a function like this.  Let the user call it prior to calling if they'd like.  Otherwise, there's no way of reproducing results.  (The same comment holds for several other functions below.)